cpp/examples/src/example_server_perf_test.cpp

```diff
@@ -0,0 +1,203 @@
+  switch (level) {
+    case foxglove::WebSocketLogLevel::Debug:
+      std::cout << "[DEBUG]" << msg << std::endl;
+      break;
+    case foxglove::WebSocketLogLevel::Info:
+      std::cout << "[INFO]" << msg << std::endl;
+      break;
+    case foxglove::WebSocketLogLevel::Warn:
+      std::cout << "[WARN]" << msg << std::endl;
+      break;
+    case foxglove::WebSocketLogLevel::Error:
+      std::cerr << "[ERROR]" << msg << std::endl;
+      break;
+    case foxglove::WebSocketLogLevel::Critical:
+      std::cerr << "[FATAL]" << msg << std::endl;
+      break;
+  }
+}
+
+void runFakeChannel(foxglove::ServerInterface<ConnectionHandle>* server,
+                    foxglove::ChannelId channelId, int sizeInBytes, double frequency) {
+  const std::vector<uint8_t> data(sizeInBytes);
+  if (frequency <= 0.0) return;
+  const uint64_t periodMs = static_cast<uint64_t>(1000.0 / frequency);
+
+  std::chrono::steady_clock::time_point lastTime = std::chrono::steady_clock::now();
+  while (!doAbort && server) {
+    // Calculate the time to sleep to maintain the desired frequency
+    const std::chrono::steady_clock::time_point currentTime = std::chrono::steady_clock::now();
+    const uint64_t elapsedMs =
+      std::chrono::duration_cast<std::chrono::milliseconds>(currentTime - lastTime).count();
+    const uint64_t sleepTimeMs = periodMs - elapsedMs;
+
+    if (sleepTimeMs > 0) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(sleepTimeMs));
+    }
+
+    // Update the lastTime for the next iteration
+    lastTime = std::chrono::steady_clock::now();
+
+    std::shared_lock<std::shared_mutex> lock(subscriptionMutex);
+    const auto& subscriberHandles = subscriptions[channelId];
+    for (const auto hdl : subscriberHandles) {
+      server->sendMessage(hdl, channelId, 0, data.data(), data.size());
+    }
+  }
+}
+
+int main(int argc, char** argv) {
+  std::string address;
+  int port;
+  std::string channelConfigFile;
+  int sendBufferLimit;
+  std::string certfile;
+  std::string keyfile;
+
+  po::options_description desc("Options");
+  desc.add_options()("address", po::value<std::string>(&address)->default_value("0.0.0.0"),
+                     "Address to bind to")("port", po::value<int>(&port)->default_value(8765),
+                                           "Port to listen to")(
+    "send-buffer-limit", po::value<int>(&sendBufferLimit)->default_value(2e7),
+    "Send buffer limit in bytes")("channel-config", po::value<std::string>(&channelConfigFile),
+                                  "Path to JSON file with custom channels in the format "
+                                  "{topic: string, frequency: number, sizeInBytes:number}[].")(
+    "compression", "Enable per-message deflate compression")(
+    "certfile", po::value<std::string>(&certfile), "Path to the certificate to use for TLS")(
+    "keyfile", po::value<std::string>(&keyfile), "Path to the private key to use for TLS")(
+    "help", "Produce help message");
```

Comment:
is this what clang-format does with boost program options? Pretty abysmal formatting ☹️ 

Could be worth wrapping in `// clang-format off ... // clang-format on`

Another idea is I think I might've worked around this in the past by replacing `desc.add_options()(...)(...)` with `auto&& add = desc.add_options(); add(...); add(...);`